packages/smart-wallet/src/walletFactory.js

```diff
@@ -142,6 +142,7 @@ export const makeAssetRegistry = assetPublisher => {
  * @param {import('@agoric/vat-data').Baggage} baggage
  */
 export const prepare = async (zcf, privateArgs, baggage) => {
+  const upgrading = baggage.has('walletsByAddress');
```

Comment:
👌 

Any ideas for catching the temptation in the future to move such a check?

@Chris-Hibbert WDYT of the zcf providing this sort of info? Perhaps even incarnationNumber?

Reply:
I think our mock bridge is lax about registration. So I expect that if we make it more strict we could reproduce this in a unit test.